packages/eservice-template-process/src/services/eserviceTemplateService.ts

```diff
@@ -2069,20 +2070,21 @@ async function updateDraftEServiceTemplateVersion(
     .exhaustive();
 
   const updatedAgreementApprovalPolicy = match(updateSeed)
-    .with({ type: "post" }, ({ seed }) =>
-      apiAgreementApprovalPolicyToAgreementApprovalPolicy(
-        seed.agreementApprovalPolicy
-      )
+    .with(
+      { type: "post" },
+      ({ seed }) =>
+        seed.agreementApprovalPolicy &&
+        apiAgreementApprovalPolicyToAgreementApprovalPolicy(
+          seed.agreementApprovalPolicy
+        )
     )
     .with({ type: "patch" }, ({ seed }) =>
-      resolvePatchValue(
-        seed.agreementApprovalPolicy === null
-          ? null
-          : apiAgreementApprovalPolicyToAgreementApprovalPolicy(
-              seed.agreementApprovalPolicy
-            ),
-        eserviceTemplateVersion.agreementApprovalPolicy
-      )
+      match(seed.agreementApprovalPolicy)
+        .with(undefined, () => eserviceTemplateVersion.agreementApprovalPolicy)
+        .with(null, () => undefined)
+        .otherwise((value) =>
+          apiAgreementApprovalPolicyToAgreementApprovalPolicy(value)
+        )
```

Comment:
Let's use the same ternary operator we used also in other places and in the resolveValue above

```suggestion
      seed.agreementApprovalPolicy === null
        ? undefined
        : seed.agreementApprovalPolicy === undefined
        ? eserviceTemplateVersion.agreementApprovalPolicy
        : apiAgreementApprovalPolicyToAgreementApprovalPolicy(
            seed.agreementApprovalPolicy
          )
```